src/presentation/controller/registration/Controller.ts

```diff
@@ -100,6 +98,42 @@ class RegistrationController extends AbstractController {
     };
   }
 
+  sendPageData(): RequestHandler {
+    return async (request: Request, response: Response, next: NextFunction) => {
+      try {
+        const access_token = this.extractAccessToken(request);
+        const pageType = this.extractPageTypeOrThrowError(request);
+        const { transactionId, submissionId } = this.extractIds(request);
+
+        const result = await this.registrationService.sendPageData(
+          { access_token },
+          transactionId,
+          submissionId,
+          pageType,
+          request.body
+        );
+
+        const registrationRouting = super.getRouting(
```

Comment:
could you swap the blocks around so that `if (result?.errors?.length) {`
is checked after the `const result` block.
If there is an error then we don't need to call `const registrationRouting = super.getRouting(`
so i think would be better to check the result errors first before calling super.getRouting(

Reply:
ah, ignore that i can see routing is needed in errors!

Reply:
we need it to get the current url - line 124